### PR TITLE
Upgrade rustls and related dependencies

### PR DESCRIPTION
**Pull request by djc**

### CODE REVIEW

File: src/authenticator.rs [hunk 1]
```diff
@@ -768,7 +768,12 @@ impl HyperClientBuilder for DefaultHyperClient {
 
     fn build_hyper_client(self) -> hyper::Client<Self::Connector> {
         #[cfg(feature = "hyper-rustls")]
-        let connector = hyper_rustls::HttpsConnector::with_native_roots();
+        let connector = hyper_rustls::HttpsConnectorBuilder::new()
+            .with_native_roots()
+            .https_only()
```
Comment (dermesser) [inline src/authenticator.rs:777]: This causes problems with the tests, as they want to hit the localhost redirect server via http. Could you add a way to not enforce https (e.g. "build_hyper_client_for_test") and replace it in `tests/tests.rs`?

**Comment by dermesser** [comment]
This looks good, save for an unintended side effect (see comment). Thank you for the PR!

**Comment by djc** [comment]
I don't see your comment?

**Comment by dermesser** [comment]
> I don't see your comment?

ah, long day hah. Submitted the in-line comment :)

**Comment by dermesser** [comment]
I took the liberty of pushing to your fork. It actually took me a while to remember the intricate ways that hyper clients are used here... the flow delegate and the authenticator use different clients. Anyway, the tests pass now at least in my local repository

**Comment by djc** [comment]
Thanks!

**Comment by djc** [comment]
Thanks!

**Pull request merged**